File: src/test/java/hwr/oop/examples/HelloWorldTest.java
package hwr.oop.examples;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

class HelloWorldTest {

    @Test
    void helloWorld_givesUsAHelloWorldString() {
        HelloWorld helloWorld = new HelloWorld("World");
        String output = helloWorld.getOutputString();
        Assertions.assertThat(output).isEqualTo("Hello World");
    }

    @Test
    void helloWorld_givesAProvidedName() {
        HelloWorld helloWorld = new HelloWorld("Tom");
        String output = helloWorld.getOutputString();
        Assertions.assertThat(output).isEqualTo("Hello Tom");
    }
}

File: examples-kotlin/lecture-01/src/test/kotlin/hwr/oop/examples/HelloWorldTest.kt
package hwr.oop.examples

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class HelloWorldTest {
	@Test
	fun `Hello World`() {
		println("Hello World!")
	}
	
	@Test
	fun `outputString is hello world, assertion with AssertJ`() {
		val output = HelloWorld.outputString()
		assertThat(output).isEqualTo("Hello World!")
	}
}